subsys/dm/dm.c

```diff
@@ -115,36 +119,44 @@ static mpsl_timeslot_request_t timeslot_request_normal = {
 	.request_type = MPSL_TIMESLOT_REQ_TYPE_NORMAL,
 	.params.normal.hfclk = MPSL_TIMESLOT_HFCLK_CFG_XTAL_GUARANTEED,
 	.params.normal.priority = MPSL_TIMESLOT_PRIORITY_HIGH,
-	.params.normal.length_us = TIMESLOT_LENGTH_US,
 };
 
 struct dm_result result;
 static mpsl_timeslot_signal_return_param_t signal_callback_return_param;
 
-static int dm_configure(void)
+static void dm_config_get(struct dm_request *dm_req, nrf_dm_config_t *dm_config)
 {
-	static nrf_dm_config_t dm_config;
+	*dm_config = NRF_DM_DEFAULT_CONFIG;
+	dm_config->access_address = dm_req->access_address;
 
-	dm_config = NRF_DM_DEFAULT_CONFIG;
-	dm_config.access_address = timeslot_ctx.curr_req.dm_req.access_address;
-
-	if (timeslot_ctx.curr_req.dm_req.role == DM_ROLE_INITIATOR) {
-		dm_config.role = NRF_DM_ROLE_INITIATOR;
-	} else if (timeslot_ctx.curr_req.dm_req.role == DM_ROLE_REFLECTOR) {
-		dm_config.role = NRF_DM_ROLE_REFLECTOR;
+	if (dm_req->role == DM_ROLE_INITIATOR) {
+		dm_config->role = NRF_DM_ROLE_INITIATOR;
 	} else {
-		return NRF_DM_STATUS_CONF_INVALID;
+		dm_config->role = NRF_DM_ROLE_REFLECTOR;
 	}
 
-	if (timeslot_ctx.curr_req.dm_req.ranging_mode == DM_RANGING_MODE_RTT) {
-		dm_config.ranging_mode = NRF_DM_RANGING_MODE_RTT;
-	} else if (timeslot_ctx.curr_req.dm_req.ranging_mode == DM_RANGING_MODE_MCPD) {
-		dm_config.ranging_mode = NRF_DM_RANGING_MODE_MCPD;
+	if (dm_req->ranging_mode == DM_RANGING_MODE_RTT) {
+		dm_config->ranging_mode = NRF_DM_RANGING_MODE_RTT;
 	} else {
-		return NRF_DM_STATUS_CONF_INVALID;
+		dm_config->ranging_mode = NRF_DM_RANGING_MODE_MCPD;
+	}
+}
+
+static uint32_t dm_proc_execute_duration(struct dm_request *dm_req)
```

Comment:
```suggestion
static uint32_t dm_proc_execute_duration_us(struct dm_request *dm_req)
```

Reply:
I see that in many other places when dealing with timing calculations _us is being used to indicate the unit

Reply:
Good point. Added the _us prefix.